Digest does not cover authentication + editorial improvements (#9392)
mdn/content: files/en-us/web/http/headers/digest/index.md

  - HTTP
  - HTTP Header
  - Digest
browser-compat: http.headers.Digest
---
{{HTTPSidebar}}

The **`Digest`** response HTTP header provides a {{Glossary("digest")}} of the _selected representation_ of the requested resource.

> Representations are different "serialization" of a particular resource that might be returned from a request: for example, the same resource might be formatted in a particular media type such as XML or JSON, localised to a particular written language or geographical region, and/or compressed or otherwise encoded for transmission.

Reviewer: Thanks!

How does "serialization" improve this/how is it more accurate than "version". I ask because when I read this I understand what version means immediately, but serialization makes me thing of over the wire formats for data transmission (i.e. packet formats). Upshot even if it is more technically accurate, it may make this harder to understand for most readers.

If you are dead set on this change, then please pluralize to "serializations".

Otherwise the changes look good to me.

Author: Hi @hamishwillee!  Ok to replace "serialization" with something else. The term "version" reminds me of caching and etags, so I'd use another term... Probably @reschke can propose a better solution in layman terms. 

Probably the best thing to do here wrt "representations" is to reuse httpbis-semantics text...

> A representation is information that is intended to reflect a past, current, or desired state of a given resource

OTOH we can just skip explaining "representation" here, and link here https://httpwg.org/http-core/draft-ietf-httpbis-semantics-latest.html#representations since it's core HTTP and people is supposed to know that already.

Reviewer: The spec definition of representation is horrible for "end users" in that if you don't know what it means already you will never be able to guess it from that text or the rest of the spec. For that reason, I'd prefer not to link to it from here, even if it is core knowledge. 

What we might do is create and link to  a glossary topic "representation" and link to the spec from - or state the spec definition. If we do this, not to self - this needs to be updated too https://developer.mozilla.org/en-US/docs/Glossary/Representation_header

For now I changed serialisation to "form" - is that a less ambiguous synonym?

Author: > if you don't know what it means already you will never be able to guess it

Agree. OTOH being a core concept, it's important to tackle it somewhere in MDN.

> form

I'd use a "representation" synonym (ex. portrayal, rendering, presentation, ..).

Reviewer: I would like to use a representation synonym but I can't find one that better than "form" or "version". I'm going to merge as is, so I can more easily create a glossary cross link for representation. I'll tag you when that's done.

> Representations are different "serialization" of a particular resource that might be returned from a request: for example, the same resource might be formatted in a particular media type such as XML or JSON, localised to a particular written language or geographical region, and/or compressed or otherwise encoded for transmission.

Author: ```suggestion
In HTTP, a resource can have multiple [representations](https://httpwg.org/http-core/draft-ietf-httpbis-semantics-latest.html#representations): might be formatted in a particular media type such as XML or JSON, localised to a particular written language or geographical region, and/or compressed or otherwise encoded for transmission.
```
